src/test/java/core/basesyntax/service/impl/FruitShopServiceImplTest.java

```diff
@@ -0,0 +1,78 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.handler.OperationHandler;
+import core.basesyntax.strategy.handler.impl.BalanceOperationHandler;
+import core.basesyntax.strategy.handler.impl.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FruitShopServiceImplTest {
+    private StorageDao storageDao;
+    private FruitShopService fruitShopService;
+    private OperationStrategy operationStrategy;
+    private Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlerMap;
+    private List<FruitTransaction> transactions;
+
+    @BeforeEach
+    void setUp() {
+        storageDao = new StorageDaoImpl();
```

Comment:
I'd recommend using Storage directly in your tests instead of storageDao instances